core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/AuthResource.scala

```diff
@@ -82,6 +76,9 @@ class AuthResource {
         // hash the plain text password
         user.setPassword(new StrongPasswordEncryptor().encryptPassword(request.password))
         userDao.insert(user)
+        if (!AmberUtils.amberConfig.getBoolean("user-sys.enabled")) {
+          user.setUid(null)
+        }
```

Comment:
what is the else case? I thought user by default should have Uid as null?

Reply:
There is no else case. UID is returned from the DB after the insert.

Reply:
ok I am confused here. I thought the intention of this if-statement is to guard the case where user-sys is not enabled on the backend (but enabled in the frontend), you want to raise an error. However a second read of this code it seems not doing this as expected.

You are modifying the register function. the current logic would insert the user to DB even if user-sys is not enabled, is that correct?

Reply:
Moved the logic to the top of the endpoint functions.